src/js/components/Tabs/Tabs.js

```diff
@@ -29,16 +42,226 @@ const Tabs = forwardRef(
+        }
+        // We are adding or subtracting 2 from amountHidden to
+        // ensure the focusIndicator is visible when navigating
+        // by keyboard
+        if (keyboard) {
+          if (amountHidden < 0) amountHidden -= 2;
+          if (amountHidden > 0) amountHidden += 2;
+        }
+        if (isSafari) {
+          headerRef.current.scrollBy({
+            left: amountHidden,
+          });
+        } else {
+          headerRef.current.scrollBy({
+            left: amountHidden,
+            behavior: 'smooth',
+          });
+        }
+
+        // wait for scroll animation to finish
+        // checks every 50 milliseconds for 1000 milliseconds
+        // if the scroll animation has finished. Most scroll
+        // animations will finish in 1000 milliseconds unless
+        // the tab name is very long.
+        if (isSafari) {
+          updateArrowState();
+        } else {
+          const checkVisible = setInterval(() => {
+            if (tabRefs[index].current && isVisible(tabRefs[index].current)) {
+              updateArrowState();
+              clearInterval(checkVisible);
+            }
+          }, 50);
+          setTimeout(() => {
+            updateArrowState();
+            clearInterval(checkVisible);
+          }, 1000);
+        }
+      },
+      [tabRefs, headerRef, isVisible, updateArrowState, isSafari],
+    );
+
+    const moveByArrowKey = (direction) => {
+      const previous = direction === 'previous';
+      let index = direction === 'previous' ? 0 : tabRefs.length - 1;
+      let scrolledToIndex;
+      const moveBy = theme.tabs.step[size] - 1 || 0;
+
+      while (
+        scrolledToIndex === undefined &&
+        ((previous && index < tabRefs.length - 1) || (!previous && index > 0))
+      ) {
+        if (
+          !isVisible(tabRefs[index].current) &&
+          ((previous && isVisible(tabRefs[index + 1].current)) ||
+            (!previous && isVisible(tabRefs[index - 1].current)))
+        ) {
+          if (previous) {
+            if (index - moveBy >= 0) {
+              scrollToIndex(index - moveBy, false);
+              scrolledToIndex = index - moveBy;
+            } else {
+              scrollToIndex(0, false);
+              scrolledToIndex = 0;
+            }
+          } else if (index + moveBy < tabRefs.length) {
+            scrollToIndex(index + moveBy, false);
+            scrolledToIndex = index + moveBy;
+          } else {
+            scrollToIndex(tabRefs.length - 1, false);
+            scrolledToIndex = tabRefs.length - 1;
+          }
+        }
+        index = previous ? index + 1 : index - 1;
+      }
+    };
+
+    useEffect(() => {
+      // if the active tab isn't visible scroll to it
+      if (
+        overflow &&
+        tabRefs &&
+        tabRefs[activeIndex].current &&
+        !isVisible(tabRefs[activeIndex].current)
+      )
+        scrollToIndex(activeIndex, true);
+      // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why do we need to disable this? More often than not, not having exhaustive dependencies has led to bugs. In the few cases where we are sure we don't want them all, we put a fairly descriptive comments explaining why.

Reply:
My reasoning for disabling this is that we only want to scroll to the active index if it is set initially, or if it is changed. If we include all the dependencies every time we tried to scroll this would force us to scroll us back so that the activeIndex is in view.

Reply:
An example of a potential issue is that `tabRefs` is re-built via `useMemo` if the `children` change. If we don't have `tabRefs` in the dependency list here, this effect will use the old one incorrectly.

Reply:
I went back and removed the `eslint-disable-next-line react-hooks/exhaustive-deps`, I realized that now that we are using `useMemo` for the tabRefs we no longer run into the issue of it trying to force us to scroll back to the activeIndex tab